snap/snapcraft.yaml

```diff
@@ -98,7 +98,9 @@ parts:
 
         echo "Compiling pyc files..."
         # This is the last step, let's now compile all our pyc files.
-        "$SNAPCRAFT_PART_INSTALL/usr/bin/python3" -m compileall -q .
+        "$SNAPCRAFT_PART_INSTALL/usr/bin/python3" -m compileall -b -q .
+        find . -type f -name '*.py' -exec rm {} \;
```

Comment:
This is an interesting one.  Does this have any effect on exceptions or any other run-time side effects?

Seems a bit risky, but mostly because I don't know the set of actions/environment changes that would trigger a rebuild (or ignoring the pyc).  I assume the fact that we're packaging all the dependencies in the snap may lower the risk, but does it eliminate it?

Although this can be satisfied easily enough in override-build as you demonstrate, I wonder if it would be useful to support a [python plugin?] flag for projects wishing to do this?

Reply:
> This is an interesting one. Does this have any effect on exceptions or any other run-time side effects?

That is indeed an issue i.e. if the relevant .py file is removed, the traceback only mentions the line number, see below
```
Traceback (most recent call last):
  File "main.py", line 10, in <module>
    thrower()
  File "main.py", line 6, in thrower
    raise Exception("OK")
Exception: OK
```
VS
```
Traceback (most recent call last):
  File "main.py", line 10, in <module>
  File "main.py", line 6, in thrower
Exception: OK
```


> Seems a bit risky, but mostly because I don't know the set of actions/environment changes that would trigger a rebuild (or ignoring the pyc).

In my understand the pyc only gets regenerated when the relevant py file changes and in case of a snap, due to its read-only nature that cannot ever happen once the package is created.